File: Sources/XMTP/ConversationExport.swift
//
//  ConversationExport.swift
//
//
//  Created by Pat Nakajima on 2/1/23.
//

import XMTPProto

enum ConversationImportError: Error {
	case invalidData
}

struct ConversationV1Export: Codable {
	var version: String
	var peerAddress: String
	var createdAt: String
}

// TODO: Make these match ConversationContainer
struct ConversationV2Export: Codable {
	var version: String
	var topic: String
	var keyMaterial: String
	var peerAddress: String
	var createdAt: String
	var context: ConversationV2ContextExport?
}

struct ConversationV2ContextExport: Codable {
	var conversationId: String
	var metadata: [String: String]
}

File: library/src/main/java/org/xmtp/android/library/ConversationExport.kt
package org.xmtp.android.library

data class ConversationV1Export(
    var version: String,
    var peerAddress: String,
    var createdAt: String,
)

data class ConversationV2Export(
    var version: String,
    var topic: String,
    var keyMaterial: String,
    var peerAddress: String,
    var createdAt: String,
    var context: ConversationV2ContextExport? = null,
)

data class ConversationV2ContextExport(
    var conversationId: String,
    var metadata: Map<String, String>,
)
